diff --git a/drivers/net/wireless/libertas/cfg.c b/drivers/net/wireless/libertas/cfg.c
index a7f1ab2..326712a 100644
--- a/drivers/net/wireless/libertas/cfg.c
+++ b/drivers/net/wireless/libertas/cfg.c
@@ -1648,11 +1648,13 @@ static int lbs_get_survey(struct wiphy *wiphy, struct net_device *dev,
 		ieee80211_channel_to_frequency(priv->channel,
 					       IEEE80211_BAND_2GHZ));
 
+#if 0
 	ret = lbs_get_rssi(priv, &signal, &noise);
 	if (ret == 0) {
 		survey->filled = SURVEY_INFO_NOISE_DBM;
 		survey->noise = noise;
 	}
+#endif
 
 	lbs_deb_leave_args(LBS_DEB_CFG80211, "ret %d", ret);
 	return ret;
