diff --git a/Makefile b/Makefile
index d71ea8e..8a36045 100644
--- a/Makefile
+++ b/Makefile
@@ -2327,6 +2327,13 @@ n516_nand_config: unconfig
 	@echo "TEXT_BASE = 0x80100000" > $(obj)board/xburst/n516/config.tmp
 	@echo "CONFIG_NAND_U_BOOT = y" >> $(obj)include/config.mk
 
+sakc_config	: unconfig
+	@echo "#define CONFIG_NAND_U_BOOT" > $(obj)include/config.h
+	@echo "Compile NAND boot image for SAKC"
+	@$(MKCONFIG) -a sakc mips xburst sakc xburst
+	@echo "TEXT_BASE = 0x80100000" > $(obj)board/xburst/sakc/config.tmp
+	@echo "CONFIG_NAND_U_BOOT = y" >> $(obj)include/config.mk
+
 #========================================================================
 # Nios
 #========================================================================
