diff --git a/include/configs/nanonote.h b/include/configs/nanonote.h
index dde5cb4..8e4f6d1 100644
--- a/include/configs/nanonote.h
+++ b/include/configs/nanonote.h
@@ -90,7 +90,7 @@
 /* Print Buffer Size */
 #define	CONFIG_SYS_MAXARGS		16	/* max number of command args*/
 
-#define CONFIG_SYS_MALLOC_LEN		896 * 1024
+#define CONFIG_SYS_MALLOC_LEN		4 * 1024 * 1024
 #define CONFIG_SYS_BOOTPARAMS_LEN	128 * 1024
 
 #define CONFIG_SYS_SDRAM_BASE		0x80000000     /* Cached addr */
diff --git a/include/configs/qi_lb60.h b/include/configs/qi_lb60.h
index 9f08c74..7ed9a1e 100644
--- a/include/configs/qi_lb60.h
+++ b/include/configs/qi_lb60.h
@@ -8,6 +8,20 @@
 //#define DEBUG
 #define CONFIG_BOOTCOMMAND	"nand read 0x80600000 0x400000 0x200000;bootm"
 #define CONFIG_BOOTARGS		"mem=32M console=tty0 console=ttyS0,57600n8 ubi.mtd=2 rootfstype=ubifs root=ubi0:rootfs rw rootwait"
+
+#define CONFIG_CMD_UBIFS
+#define CONFIG_CMD_UBI
+#define CONFIG_MTD_PARTITIONS
+#define CONFIG_MTD_DEVICE
+#define CONFIG_CMD_MTDPARTS
+#define CONFIG_CMD_UBI
+#define CONFIG_CMD_UBIFS
+#define CONFIG_LZO
+#define CONFIG_RBTREE
+
+#define MTDIDS_DEFAULT		"nand0=jz4740-nand"
+#define MTDPARTS_DEFAULT	"mtdparts=jz4740-nand:4M@0(uboot)ro,4M@4M(kernel)ro,512M@8M(rootfs)ro,-(data)ro"
+
 #define CONFIG_EXTRA_ENV_SETTINGS \
 	"bootcmdfromsd=mmc init; ext2load mmc 0 0x80600000 /boot/uImage; bootm;\0" \
 	"bootargsfromsd=mem=32M console=tty0 console=ttyS0,57600n8 rootfstype=ext2 root=/dev/mmcblk0p1 rw rootwait\0" \
@@ -17,8 +31,8 @@
 	"bootargsf2=mem=32M console=tty0 console=ttyS0,57600n8 rootfstype=ext2 root=/dev/mmcblk0p2 rw rootwait\0" \
 	"bootcmdf3=mmc init; ext2load mmc 0:3 0x80600000 /boot/uImage; bootm;\0" \
 	"bootargsf3=mem=32M console=tty0 console=ttyS0,57600n8 rootfstype=ext2 root=/dev/mmcblk0p3 rw rootwait\0" \
-	"bootcmdf4=mmc init; ext2load mmc 0:4 0x80600000 /boot/uImage; bootm;\0" \
-	"bootargsf4=mem=32M console=tty0 console=ttyS0,57600n8 rootfstype=ext2 root=/dev/mmcblk0p4 rw rootwait\0"
+	"bootcmdf4=mtdparts default;ubi part rootfs;ubifsmount rootfs;ubifsload 0x80600000 /boot/uImage; bootm;\0" \
+	"bootargsf4=mem=32M console=tty0 console=ttyS0,57600n8 ubi.mtd=2 rootfstype=ubifs root=ubi0:rootfs rw rootwait"
 
 /* SDRAM paramters */
 #define SDRAM_BW16		1	/* Data bus width: 0-32bit, 1-16bit */
