Index: u-boot-2010.06/include/configs/nanonote.h
===================================================================
--- u-boot-2010.06.orig/include/configs/nanonote.h	2011-08-29 12:59:47.520570290 +0200
+++ u-boot-2010.06/include/configs/nanonote.h	2011-08-29 13:55:48.958207218 +0200
@@ -99,6 +99,8 @@
 #define CONFIG_SYS_MEMTEST_START	0x80100000
 #define CONFIG_SYS_MEMTEST_END		0x80800000
 
+#define CONFIG_SILENT_CONSOLE		1	/* Enable silent console */
+
 /*
  * Environment
  */
Index: u-boot-2010.06/common/console.c
===================================================================
--- u-boot-2010.06.orig/common/console.c	2011-08-29 13:18:53.987163122 +0200
+++ u-boot-2010.06/common/console.c	2011-08-29 14:15:26.303638381 +0200
@@ -628,6 +628,14 @@
 
 	gd->flags |= GD_FLG_DEVINIT;	/* device initialization completed */
 
+#ifdef CONFIG_SILENT_CONSOLE
+	/* Check one more time the contents of the silent environment
+	 * variable, because if the environment is loaded from NAND it was
+	 * not available when console_init_f() was called */
+	if (getenv("silent") != NULL)
+		gd->flags |= GD_FLG_SILENT;
+#endif
+
 	stdio_print_current_devices();
 
 #ifdef CONFIG_SYS_CONSOLE_ENV_OVERWRITE
@@ -703,6 +711,14 @@
 
 	gd->flags |= GD_FLG_DEVINIT;	/* device initialization completed */
 
+#ifdef CONFIG_SILENT_CONSOLE
+	/* Check one more time the contents of the silent environment
+	 * variable, because if the environment is loaded from NAND it was
+	 * not available when console_init_f() was called */
+	if (getenv("silent") != NULL)
+		gd->flags |= GD_FLG_SILENT;
+#endif
+
 	stdio_print_current_devices();
 
 	/* Setting environment variables */
