diff --git a/include/configs/qi_lb60.h b/include/configs/qi_lb60.h
index 7ed9a1e..9b1a2f7 100644
--- a/include/configs/qi_lb60.h
+++ b/include/configs/qi_lb60.h
@@ -6,7 +6,7 @@
 #define CONFIG_QI_LB60 1
 
 //#define DEBUG
-#define CONFIG_BOOTCOMMAND	"nand read 0x80600000 0x400000 0x200000;bootm"
+#define CONFIG_BOOTCOMMAND	"nand read 0x80600000 0x400000 0x280000;bootm"
 #define CONFIG_BOOTARGS		"mem=32M console=tty0 console=ttyS0,57600n8 ubi.mtd=2 rootfstype=ubifs root=ubi0:rootfs rw rootwait"
 
 #define CONFIG_CMD_UBIFS
