From 65716cc0ff058001a7cbcb77a3eababc8949840b Mon Sep 17 00:00:00 2001
From: Vasily Khoruzhick <anarsoul@gmail.com>
Date: Mon, 4 Jul 2011 12:36:39 +0300
Subject: [PATCH 07/18] Z2: add sysrq key

Signed-off-by: Vasily Khoruzhick <anarsoul@gmail.com>
---
 arch/arm/mach-pxa/z2.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-pxa/z2.c b/arch/arm/mach-pxa/z2.c
index 2e5aadc..a7aafe3 100644
--- a/arch/arm/mach-pxa/z2.c
+++ b/arch/arm/mach-pxa/z2.c
@@ -361,7 +361,7 @@ static unsigned int z2_matrix_keys[] = {
 	KEY(5, 0, KEY_END),
 	KEY(6, 0, KEY_KPPLUS),
 
-	KEY(0, 1, KEY_HOME),
+	KEY(0, 1, KEY_SYSRQ),
 	KEY(1, 1, KEY_Q),
 	KEY(2, 1, KEY_I),
 	KEY(3, 1, KEY_G),
-- 
1.7.1

