From 39f1d0ea341360f9bdf570e547e2d55077430f7e Mon Sep 17 00:00:00 2001
From: Vasily Khoruzhick <anarsoul@gmail.com>
Date: Tue, 31 May 2011 20:47:00 +0300
Subject: [PATCH 05/18] ARM: PXA: Zipit Z2: Add poweroff function

Signed-off-by: Vasily Khoruzhick <anarsoul@gmail.com>
---
 arch/arm/mach-pxa/z2.c |   14 ++++++++++++++
 1 files changed, 14 insertions(+), 0 deletions(-)

Index: linux-2.6.37.6/arch/arm/mach-pxa/z2.c
===================================================================
--- linux-2.6.37.6.orig/arch/arm/mach-pxa/z2.c	2011-08-17 23:15:59.440699003 +1000
+++ linux-2.6.37.6/arch/arm/mach-pxa/z2.c	2011-08-17 23:18:57.270699001 +1000
@@ -39,6 +39,7 @@
 #include <mach/mmc.h>
 #include <plat/pxa27x_keypad.h>
 #include <mach/pxa2xx_spi.h>
+#include <mach/pm.h>
 
 #include <plat/i2c.h>
 
@@ -683,6 +685,17 @@
 static inline void z2_pmic_init(void) {}
 #endif
 
+#ifdef CONFIG_PM
+static void z2_power_off(void)
+{
+	PSPR = 0x0;
+	local_irq_disable();
+	pxa27x_cpu_suspend(PWRMODE_DEEPSLEEP);
+}
+#else
+#define z2_power_off   NULL
+#endif
+
 /******************************************************************************
  * Machine init
  ******************************************************************************/
@@ -704,6 +717,8 @@
 	z2_leds_init();
 	z2_keys_init();
 	z2_pmic_init();
+
+	pm_power_off = z2_power_off;
 }
 
 MACHINE_START(ZIPIT2, "Zipit Z2")
